pkg/scheduler/extender.go

```diff
@@ -246,18 +246,18 @@ func convertToMetaVictims(
 // unresolvable.
 func (h *HTTPExtender) Filter(
 	pod *v1.Pod,
-	nodes []*v1.Node,
-) (filteredList []*v1.Node, failedNodes, failedAndUnresolvableNodes extenderv1.FailedNodesMap, err error) {
+	nodes []*framework.NodeInfo,
```

Comment:
Undo any changes in extenders.

Reply:
Hmm, if we do not change the extender. We'll always get a list of node as candidates in filter, right?
https://github.com/kubernetes/kubernetes/blob/0c645922edcc06adff43c70c02fb56751364bbb5/pkg/scheduler/schedule_one.go#L503
And if we just pass this node list to score extension, it's just the same as we don't change at all.
Are you suggesting in L503 here, we pass a node list from nodeInfos and get a result list, and use this result list to filter out some nodeInfos from previous result list? Like this?
```
	nodeInfos, err := sched.findNodesThatPassFilters(ctx, fwk, state, pod, &diagnosis, nodes)
	nds, err = findNodesThatPassExtenders(ctx, sched.Extenders, pod, framework.GetNodes(nodeInfos), diagnosis.NodeToStatusMap)

	// return nodeInfo if it's node is in "nds"
	return getNodeInfos(nodeInfos,nds), diagnosis, nil
```
However this also could be quite expensive in a big cluster.
I can't think of a better way to deal with it🤔️.

Reply:
I see.

Let's leave the changes here and work on the other comments.